pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -106,6 +108,10 @@ func newEndpointSlice(endpoints *corev1.Endpoints, ports []discovery.EndpointPor
 	epSlice.Labels[discovery.LabelServiceName] = endpoints.Name
 	epSlice.Labels[discovery.LabelManagedBy] = controllerName
 
+	for annotation, val := range endpoints.Annotations {
+		epSlice.Annotations[annotation] = val
```

Comment:
Should the `endpoints.kubernetes.io/last-change-trigger-time` annotation be skipped here? (see https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpointslice/utils.go#L284)

Reply:
@robscott @freehan what do you think?

Reply:
That's a good question. I think we should _probably_ remove that annotation. There is always a chance that whoever is writing the custom endpoints this controller is mirroring is also setting this annotation correctly. I think there's a somewhat higher probability that this annotation will get stuck with a stale value that is never updated after a Service transitions to an empty selector. The [code that we're relying on for TriggerTimeTracker](https://github.com/kubernetes/kubernetes/blob/d70ee902fddc682863a3cc4f0d8eac0223ebf70b/pkg/controller/util/endpoint/trigger_time_tracker.go#L84) just doesn't make sense for custom endpoints, so I think it would be good to omit the annotation.